### ISSUE TITLE
[wasm-built] Inbuilt instrumentation support with threads enabled

### ISSUE THREAD

**Issue by ravisumit33** [emscripten-core/emscripten#22420]
Currently, emscripten instruments wasm binaries without using `--in-memory` or `--in-secondary-memory` wasm-split options when threads are enabled [here](https://github.com/emscripten-core/emscripten/blob/main/tools/link.py#L466). Even [docs](https://emscripten.org/docs/optimizing/Module-Splitting.html#multithreaded-programs) mention using `--in-memory` for multithreaded programs. I have few questions regarding this:

1. Do we need to re-instrument the wasm binary ourselves for multithreaded programs? Or has emscripten planned to provide support for this?
2. Docs also suggest to use `-sGLOBAL_BASE` flag with `--in-memory` wasm-split option. What about `--in-secondary-memory` option? Do we still need to use that flag if we use the later option?

**Comment by ravisumit33** [comment]
@tlively @sbc100 Thoughts?

**Comment by tlively** [comment]
Yes, right now you would have to re-instrument the binary yourself to use one of those options. Ideally we would support `--in-secondary-memory` to avoid having to mess with `-sGLOBAL_BASE`, but Emscripten does not currently know how to provide the second memory when instantiating the module, so your best bet is to use `--in-memory` and set an appropriate `-sGLOBAL_BASE`.

**Comment by ravisumit33** [comment]
@tlively I tried re-instrumenting the wasm again using following command:

```sh
wasm-split -all --instrument --in-memory test.wasm.orig -o test.wasm
```

Some of the ourworkflows went well in profiling but some complex workflows started giving weird memory access out of bound errors. I tried building debug build (-g) but weirdingly not a single workflow went well and some random errors came which were not there in single threaded build and are not related. My guess is there is some memory corruption/bug related to `--in-memory` wasm-split flag and/or `-sGLOBAL_BASE` emscripten flag. What are your views?
